src/cli/calculate.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  CalculationsByResultType,
+  CalculationsByGroupId,
+  ProcessDescription,
+  ResultsByGroupId,
+  ResultType,
+} from './types';
+import { performAllCalculations } from './util/calculate';
+import { writeFile } from './util/write';
+
+const calculate = (resultsByType: (ResultsByGroupId & { type: ResultType })[]) => {
+  /**
+   * Calculate the max, min, median, and mean values
+   * of the previously formatted outputs, and group the results
+   * per result type (baseline vs. current).
+   */
+  const calculationsByResultType = resultsByType.reduce(
```

Comment:
I've read recently, and am now convinced that reducing with spreads is not ideal, and maybe a Object.fromEntries() is a better way to go about the same thing (better perf, easier to follow). Mind here it's a bit of a nit, but up to you if you think the change makes ssense.

Reply:
nice love it